flink-end-to-end-tests/test-scripts/test_connectors_hive.sh

```diff
@@ -0,0 +1,170 @@
+    # to run
+    nm_running=`docker exec -it master bash -c "yarn node -list" | grep RUNNING | wc -l`
+    start_time=$(date +%s)
+    while [[ "$nm_running" -lt "2" ]]; do
+        current_time=$(date +%s)
+        time_diff=$((current_time - start_time))
+
+        if [ $time_diff -ge $MAX_RETRY_SECONDS ]; then
+            return 1
+        else
+            echo "We only have $nm_running NodeManagers up. We have been trying for $time_diff seconds, retrying ..."
+            sleep 1
+        fi
+        nm_running=`docker exec -it master bash -c "yarn node -list" | grep RUNNING | wc -l`
+    done
+
+    return 0
+}
+echo "Building Hadoop hive Docker container"
+until docker build --build-arg HADOOP_VERSION=2.8.4 --build-arg HIVE_VERSION=2.3.4 \
+    -f $END_TO_END_DIR/test-scripts/docker-hive-hadoop-cluster/Dockerfile \
+    -t flink/flink-hadoop-hive-cluster:latest \
+    $END_TO_END_DIR/test-scripts/docker-hive-hadoop-cluster/;
+do
+    # with all the downloading and ubuntu updating a lot of flakiness can happen, make sure
+    # we don't immediately fail
+    echo "Something went wrong while building the Docker image, retrying ..."
+    sleep 2
+done
+
+if ! retry_times $CLUSTER_SETUP_RETRIES 0 start_hadoop_cluster_and_hive; then
+    echo "ERROR: Could not start hadoop cluster and hive. Aborting..."
+    exit 1
+fi
+
+mkdir -p $FLINK_TARBALL_DIR
+tar czf $FLINK_TARBALL_DIR/$FLINK_TARBALL -C $(dirname $FLINK_DIR) .
+
+docker cp $FLINK_TARBALL_DIR/$FLINK_TARBALL master:/home/hadoop-user/
+docker cp $END_TO_END_DIR/flink-hive-test/target/testHive.jar master:/home/hadoop-user/
+
+# now, at least the container is ready
+docker exec -it master bash -c "tar xzf /home/hadoop-user/$FLINK_TARBALL --directory /home/hadoop-user/"
+
+# minimal Flink config, bebe
+FLINK_CONFIG=$(cat << END
+slot.request.timeout: 120000
+containerized.heap-cutoff-min: 100
+END
+)
+docker exec -it master bash -c "echo \"$FLINK_CONFIG\" > /home/hadoop-user/$FLINK_DIRNAME/conf/flink-conf.yaml"
+
+echo "Flink config:"
+docker exec -it master bash -c "cat /home/hadoop-user/$FLINK_DIRNAME/conf/flink-conf.yaml"
+
+function copy_and_show_logs {
+    mkdir -p $TEST_DATA_DIR/logs
+    echo "Hadoop logs:"
+    docker cp master:/var/log/hadoop/* $TEST_DATA_DIR/logs/
+    for f in $TEST_DATA_DIR/logs/*; do
+        echo "$f:"
+        cat $f
+    done
+    echo "Docker logs:"
+    docker logs master
+
+    echo "Flink logs:"
+    application_id=`docker exec -it master bash -c "yarn application -list -appStates ALL" | grep "Flink session cluster" | awk '{print \$1}'`
+    echo "Application ID: $application_id"
+    docker exec -it master bash -c "yarn logs -applicationId $application_id"
+}
+
+start_time=$(date +%s)
+if docker exec -it master bash -c "export HADOOP_CLASSPATH=\`hadoop classpath\` && \
+   /home/hadoop-user/$FLINK_DIRNAME/bin/flink run -m yarn-cluster -ys 1 -ytm 1000 -yjm 1000 \
+   -p 1 /home/hadoop-user/testHive.jar";
```

Comment:
maybe we should also specify the class to run, so that it's easier to add new test cases